Implement CLI functionality

jmcph4 opened the pull request:
# Description #
See #1. 

# Rationale #
See #1. 

# Acceptance Criteria #
 - [x] Invocations of `rainyday` fail and display usage information
 - [x] Invocations of `rainyday --help` succeed and display help information
 - [x] Invocations of `rainyday --version` succeed and display version information

jmcph4 committed:
Add `clap` dependency

This commit adds the `clap` crate as a dependency to enable parsing of
command-line arguments.

jmcph4 committed:
Add Rustfmt configuration

This commit adds a configuration file for Rustfmt.

jmcph4 committed:
Implement command-line parsing

This commit defines the command-line interface via `clap`.

alecdwm commented on Cargo.toml:
```diff
@@ -7,3 +7,5 @@ edition = "2018"
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
+clap = "2.33.3"
```
Clap v3's [derive macro is pretty cool](https://github.com/clap-rs/clap#using-derive-macros) if you wanted to try it out

jmcph4 replied:
Noted, see 5eff499.

alecdwm commented on src/main.rs:
```diff
@@ -1,3 +1,8 @@
 fn main() {
-    println!("Hello, world!");
+    let _matches: ArgMatches = App::new("rainyday")
+        .version("0.0.1")
```
You can pull this from the cargo.toml like so:

`.version(crate_version!())`

[example here](https://github.com/clap-rs/clap/blob/HEAD/examples/09_auto_version.rs)

jmcph4 replied:
Very cool! See 1ea28dd.

alecdwm reviewed (commented):
Added some comments

jmcph4 committed:
Use `crate_version!` macro to enforce DRY code

This commit uses the `crate_version!` macro, at @alecdwm's direction.

jmcph4 committed:
Switch to Clap v3 derives for arg parsing

This commit switches the CLI argument parsing code from Clap 2.x to
Clap 3.x using the derive macros.

Once again, thanks to @alecdwm.

alecdwm commented on Cargo.toml:
```diff
@@ -7,3 +7,5 @@ edition = "2018"
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
+clap = { git = "https://github.com/clap-rs/clap/" }
```
They have a prerelease on crates.io!
`clap = "3.0.0-beta.2"`

jmcph4 replied:
Good catch! Fixed in 19cb46f.

alecdwm reviewed (commented):
Looks great! One more comment

jmcph4 committed:
Point Cargo at the official pre-release for Clap 3.x

This commit uses the official Crates.io pre-release for Clap 3.x,
rather than the GitHub repository.
